Assets/AirConsole/scripts/AirConsole.cs

```diff
@@ -1788,8 +1788,20 @@ public class NotReadyException: SystemException {
             Debug.Log("onLaunchApp");
 			string gameId = (string)msg ["game_id"];
 			string gameVersion = (string)msg ["game_version"];
+			bool noThreadSleepOnGameChange = false;
+
+			if (msg["no_thread_sleep_on_game_change"] != null) {
+				noThreadSleepOnGameChange = msg.SelectToken("no_thread_sleep_on_game_change").Value<bool>();
+			}
+
 			if (gameId != Application.identifier || gameVersion != AirConsole.instance.androidTvGameVersion) {
+				// Quit the Unity Player first and give it the time to close are threads
+				Application.Quit();
```

Comment:
I would move this in the next if statement and at the end add an if statement for the opposite path